icache.go

```diff
@@ -19,3 +20,22 @@ type Cache interface {
 
 	Close() error
 }
+
+// SimpleCache is a cache interface that allows any type of key and value
+// as long as the key is hashable and the value is serializable.
+//
+//go:generate mockgen -destination=icache_mock.go -package=cache github.com/InjectiveLabs/injective-cache SimpleCache
+type SimpleCache interface {
+	Set(ctx context.Context, key interface{}, value interface{}) (err error)
+	SetWithTTL(ctx context.Context, key interface{}, value interface{}, ttl time.Duration) (err error)
+	Get(ctx context.Context, key interface{}, value interface{}) (err error)
+	Del(ctx context.Context, keys ...interface{}) (err error)
+	Clear(ctx context.Context) (err error)
+}
```

Comment:
I wonder if `SimpleCache` should really be an interface. This is just a Cache with TTL.
Couldn't we just have some kind of `TTLCache` interface that embeds the `Cache` one? And we implement the `TTLCache` interface on `TypedLibCache`.

We would need to change the `Cache` interface a bit to accept `any` instead of strings/[]bytes but thats no big deal. Whoever update the `injective-cache` module can change the integration really fast.

Or is there any other reason or thing that I am missing?

Reply:
the problem with this approach is that is breaking the old interface, so if we upgrade in repos with heavy usage we need to change all the places where is being used, we should tag a new version then, since this repo being public

Reply:
Upgrading interfaces dont really bother me. Code should be evolving as we grow and this is why we got versioning. Still, I get it if you don't want to do the change.